h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -1748,100 +1754,120 @@ private int getMovePriority(Chunk chunk) {
     private void compactMoveChunks(Iterable<Chunk> move) {
         assert storeLock.isHeldByCurrentThread();
         if (move != null) {
-            long leftmostBlock = Long.MAX_VALUE;
-            for (Chunk chunk : move) {
-                leftmostBlock = Math.min(leftmostBlock, chunk.block);
-            }
-
+            assert lastChunk != null;
             // this will ensure better recognition of the last chunk
             // in case of power failure, since we are going to move older chunks
             // to the end of the file
             writeStoreHeader();
             sync();
-            for (Iterator<Chunk> iter = move.iterator(); iter.hasNext(); ) {
-                Chunk chunk = iter.next();
-                if (fileStore.predictAllocation(chunk.len) >= leftmostBlock) {
-                    break;
-                }
-                moveChunk(chunk, false);
-                iter.remove();
-            }
 
+            long leftmostBlock = Long.MAX_VALUE;
+            for (Chunk chunk : move) {
+                leftmostBlock = Math.min(leftmostBlock, chunk.block);
+            }
```

Comment:
The for statement is unnecessary for leftmostBlock , because of the move iterator has been in order. The leftmostBlock  actually is the block of the first chunk in move iterator.

Reply:
Yep, it's redundant now, removing